### PR TITLE
[NO-ISSUE] Use apache-release profile when doing a release build

### PR DESCRIPTION
**Pull request by baldimir**
Adds gpg setup and apache-release Maven profile to the release build.

### CODE REVIEW

File: .ci/jenkins/Jenkinsfile.deploy [hunk 1]
```diff
@@ -110,13 +110,29 @@ pipeline {
                         } else {
                             installOrDeploy = 'install'
                         }
+
+                        def mavenCommand = getMavenCommand().withProperty('maven.test.failure.ignore', true)
+                           .withOptions(env.BUILD_MVN_OPTS_CURRENT ? [ env.BUILD_MVN_OPTS_CURRENT ] : [])
+                           .withOptions(env.KOGITO_APPS_BUILD_MVN_OPTS ? [ env.KOGITO_APPS_BUILD_MVN_OPTS ] : [])
+                           .skipTests(skipTests)
+                           .withSettingsXmlFile(MAVEN_SETTINGS_FILE)
```
Comment (jstastny-cz) [inline .ci/jenkins/Jenkinsfile.deploy]: again the settings file need to be accessed only in configFileProvider

**Review by jstastny-cz — CHANGES_REQUESTED** [review]

**Comment by kie-ci3** [comment]
**PR job** `#1` was: **UNSTABLE**
Possible explanation: This should be test failures


<details>
<summary><b>Reproducer</b></summary>



build-chain build full_downstream  -f 'https://raw.githubusercontent.com/${AUTHOR:apache}/incubator-kie-kogito-pipelines/${BRANCH:main}/.ci/buildchain-config-pr-cdb.yaml' -o 'bc' -p apache/incubator-kie-kogito-apps -u https://github.com/apache/incubator-kie-kogito-apps/pull/2062 --skipParallelCheckout

NOTE: To install the build-chain tool, please refer to https://github.com/kiegroup/github-action-build-chain#local-execution

</details>


Please look here: https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-apps-pr/job/PR-2062/1/display/redirect

**Test results:**
- PASSED: 1610
- FAILED: 4

Those are the test failures: 
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-apps-pr/job/PR-2062/1/testReport/org.kie.kogito.index.infinispan/ProcessDataIndexInfinispanHttpIT/testProcessInstanceEvents/">org.kie.kogito.index.infinispan.ProcessDataIndexInfinispanHttpIT.testProcessInstanceEvents</a></summary>
1 expectation failed.<br/>JSON path data.UserTaskInstances[0].potentialGroups[0] doesn't match.<br/>Expected: managers<br/>  Actual: null<br/>
</details>
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-apps-pr/job/PR-2062/1/testReport/org.kie.kogito.index.mongodb/ProcessDataIndexMongoDBHttpIT/testProcessInstanceEvents/">org.kie.kogito.index.mongodb.ProcessDataIndexMongoDBHttpIT.testProcessInstanceEvents</a></summary>
1 expectation failed.<br/>JSON path data.UserTaskInstances[0].potentialGroups[0] doesn't match.<br/>Expected: managers<br/>  Actual: null<br/>
</details>
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-apps-pr/job/PR-2062/1/testReport/org.kie.kogito.index.mongodb/ProcessDataIndexMongoDBKafkaIT/testProcessInstanceEvents/">org.kie.kogito.index.mongodb.ProcessDataIndexMongoDBKafkaIT.testProcessInstanceEvents</a></summary>
1 expectation failed.<br/>JSON path data.UserTaskInstances[0].potentialGroups[0] doesn't match.<br/>Expected: managers<br/>  Actual: null<br/>
</details>
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-apps-pr/job/PR-2062/1/testReport/org.kie.kogito.index/ProcessDataIndexMongoDBIT/testProcessInstanceEvents/">org.kie.kogito.index.ProcessDataIndexMongoDBIT.testProcessInstanceEvents</a></summary>
1 expectation failed.<br/>JSON path data.UserTaskInstances[0].potentialGroups[0] doesn't match.<br/>Expected: managers<br/>  Actual: null<br/>
</details>

File: .ci/jenkins/Jenkinsfile.deploy (hunk 1 above)
Comment (baldimir) [reply .ci/jenkins/Jenkinsfile.deploy]: Fixed.

**Review by jstastny-cz — APPROVED** [review]
LGTM.

**Comment by kie-ci3** [comment]
**PR job** `#4` was: **UNSTABLE**
Possible explanation: This should be test failures


<details>
<summary><b>Reproducer</b></summary>



build-chain build full_downstream  -f 'https://raw.githubusercontent.com/${AUTHOR:apache}/incubator-kie-kogito-pipelines/${BRANCH:main}/.ci/buildchain-config-pr-cdb.yaml' -o 'bc' -p apache/incubator-kie-kogito-apps -u https://github.com/apache/incubator-kie-kogito-apps/pull/2062 --skipParallelCheckout

NOTE: To install the build-chain tool, please refer to https://github.com/kiegroup/github-action-build-chain#local-execution

</details>


Please look here: https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-apps-pr/job/PR-2062/4/display/redirect

**Test results:**
- PASSED: 1611
- FAILED: 3

Those are the test failures: 
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-apps-pr/job/PR-2062/4/testReport/org.kie.kogito.index.mongodb/ProcessDataIndexMongoDBHttpIT/testProcessInstanceEvents/">org.kie.kogito.index.mongodb.ProcessDataIndexMongoDBHttpIT.testProcessInstanceEvents</a></summary>
1 expectation failed.<br/>JSON path data.UserTaskInstances[0].potentialGroups[0] doesn't match.<br/>Expected: managers<br/>  Actual: null<br/>
</details>
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-apps-pr/job/PR-2062/4/testReport/org.kie.kogito.index.mongodb/ProcessDataIndexMongoDBKafkaIT/testProcessInstanceEvents/">org.kie.kogito.index.mongodb.ProcessDataIndexMongoDBKafkaIT.testProcessInstanceEvents</a></summary>
1 expectation failed.<br/>JSON path data.UserTaskInstances[0].potentialGroups[0] doesn't match.<br/>Expected: managers<br/>  Actual: null<br/>
</details>
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-apps-pr/job/PR-2062/4/testReport/org.kie.kogito.index/ProcessDataIndexMongoDBIT/testProcessInstanceEvents/">org.kie.kogito.index.ProcessDataIndexMongoDBIT.testProcessInstanceEvents</a></summary>
1 expectation failed.<br/>JSON path data.UserTaskInstances[0].potentialGroups[0] doesn't match.<br/>Expected: managers<br/>  Actual: null<br/>
</details>

**Pull request merged**